src/renderer/components/+preferences/preferences-navigation/extensions-preferences-navigation-item.injectable.ts

```diff
@@ -3,49 +3,54 @@
 import { computed } from "mobx";
-import extensionsPreferenceItemsInjectable from "../extension-preference-items.injectable";
 import extensionPreferencesRouteInjectable from "../../../../common/front-end-routing/routes/preferences/extension/extension-preferences-route.injectable";
-import navigateToPreferenceTabInjectable from "./navigate-to-preference-tab.injectable";
+import navigateToExtensionPreferencesInjectable from "../../../../common/front-end-routing/routes/preferences/extension/navigate-to-extension-preferences.injectable";
+import { extensionRegistratorInjectionToken } from "../../../../extensions/extension-loader/extension-registrator-injection-token";
+import type { LensRendererExtension } from "../../../../extensions/lens-renderer-extension";
+import currentPathParametersInjectable from "../../../routes/current-path-parameters.injectable";
+import routeIsActiveInjectable from "../../../routes/route-is-active.injectable";
+import { preferenceNavigationItemInjectionToken } from "./preference-navigation-items.injectable";
 
-const extensionsPreferencesNavigationItemInjectable = getInjectable({
+const extensionPreferencesNavigationItemRegistratorInjectable = getInjectable({
   id: "extension-preferences-navigation-item",
 
   instantiate: (di) => {
-    const preferenceItems = di.inject(
-      extensionsPreferenceItemsInjectable,
-    );
-
-    const navigateToPreferenceTab = di.inject(
-      navigateToPreferenceTabInjectable,
-    );
-
-    const route = di.inject(
-      extensionPreferencesRouteInjectable,
-    );
-
-    const routeIsActive = di.inject(
-      routeIsActiveInjectable,
-      route,
-    );
-
-    return {
-      id: "extensions",
-      label: "Extensions",
-      navigate: navigateToPreferenceTab(route),
-      isActive: routeIsActive,
-
-      isVisible: computed(
-        () => preferenceItems.get().length > 0,
-      ),
-
-      orderNumber: 70,
+    return (ext) => {
+      const extension = ext as LensRendererExtension;
+      const navigateToExtensionPreferences = di.inject(
+        navigateToExtensionPreferencesInjectable,
+      );
+      const extensionHasPreferences = extension.appPreferences.length > 0;
+      const extensionHasGeneralPreferences = extension.appPreferences.some(preferences =>
+        !preferences.showInPreferencesTab,
+      );
+      const isVisible = computed(() => extensionHasPreferences && extensionHasGeneralPreferences);
+      const extensionRoute = di.inject(extensionPreferencesRouteInjectable);
+      const pathParameters = di.inject(currentPathParametersInjectable);
+      const routeIsActive = di.inject(routeIsActiveInjectable, extensionRoute);
+
+      const isActive = computed(() => routeIsActive.get() && pathParameters.get().extensionId === extension.sanitizedExtensionId);
+
+      const extensionInjectable = getInjectable({
+        id: `extension-preferences-navigation-item-${extension.sanitizedExtensionId}`,
```

Comment:
For now, need to know stuff. In line 19. there is second argument which is `extensionInstallationCount` Let's add that to the `id` in line 36. This is to allow enabling, disabling and re-enabling extension. 

This is something that we'll get rid of soon being absurd technical complexity. :)

Reply:
https://github.com/lensapp/lens/pull/5662 This PR removes the `extensionInstallationCount` for good.